diff --git a/src/main/java/org/reaktivity/nukleus/tls/internal/stream/ServerStreamFactory.java b/src/main/java/org/reaktivity/nukleus/tls/internal/stream/ServerStreamFactory.java
index e0fee7c..dbac467 100644
--- a/src/main/java/org/reaktivity/nukleus/tls/internal/stream/ServerStreamFactory.java
+++ b/src/main/java/org/reaktivity/nukleus/tls/internal/stream/ServerStreamFactory.java
@@ -1017,7 +1017,7 @@ public final class ServerStreamFactory implements StreamFactory
                             networkSlotOffset = totalBytesRemaining;
                             break loop;
                         default:
-                            networkSlotOffset = 0;
+                            networkSlotOffset = inNetByteBuffer.remaining();
                             statusHandler.accept(handshakeStatus, this::updateNetworkWindow);
                             break;
                         }
